keras/utils/dot_utils.py

```diff
@@ -0,0 +1,41 @@
+import pydot
+from keras.layers.core import Merge
+from keras.models import Model
+from collections import Counter
+
+class Grapher(object):
+
+    def __init__(self):
+        self.names = {}
+        self.class_counts = Counter()
+
+    def get_name(self, model):
+        if hasattr(model, 'name'):
+            return model.name
+        cls = model.__class__.__name__
```

Comment:
`cls` is a reserved keyword, it would be better to use a different variable name.